### PR TITLE
Saml metadata xml upload

### LINKED ISSUE #26354: SAML configuration via XML upload
**Feature**
Currently, for Saml SSO configuration, we upload, Idp related fields manually.
But we can do this using metadata XML from customer, which is simpler.

Choose XML -> Validate -> Save Configuration

### PR DESCRIPTION
**Pull request by Rohit0301**
### Describe your changes:

Fixes #26354 

I worked on ... because ...

https://github.com/user-attachments/assets/82c3eca2-ae47-4311-af76-f6ca2810ad5c

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

----
## Summary by Gitar

- **SAML metadata XML upload feature:**
  - Added file upload UI component with drag-and-drop support for SAML IdP metadata XML files
  - Auto-fills `entityId`, `ssoLoginUrl`, and `idpX509Certificate` fields from parsed metadata
  - Displays success/error status cards with option to re-upload
- **Parsing logic:**
  - Implemented `parseSamlMetadataXml` utility supporting Azure AD, Okta, Auth0, Google Workspace metadata formats
  - Enforces required X.509 certificate; prefers `use="signing"` certificates
- **Tests & localization:**
  - Added 373-line test suite covering valid/invalid metadata, binding preferences, edge cases
  - Added E2E tests for upload flow, success/error states, file re-upload
  - Updated 18 language files with UI strings

<sub>This will update automatically on new commits.</sub>

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx [hunk 1]
```diff
@@ -96,6 +99,57 @@ import {
...
+}: MetadataUploadStatusCardProps) => {
+  const { t } = useTranslation();
+  const isSuccess = status === 'success';
+
+  return (
+    <div className="flex items-center justify-between p-xs metadata-upload-status-container">
+      <div className="flex items-center gap-2">
+        <div
+          className={classNames(
+            'flex-shrink flex items-center justify-center rounded-full w-6 h-6',
```
[the remark this answers is not in the document]
Comment (Rohit0301) [reply openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx:122]: flex-shrink class is defined in the code in css file.

[the remark this answers is not in the document]
Comment (Rohit0301) [reply openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx:122]: @gitar-bot please check

File: openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx [hunk 2]
```diff
@@ -249,6 +310,78 @@ const SSOConfigurationFormRJSF = ({
...
   }, []);
 
+  const handleMetadataFileUpload = useCallback(
+    (files: FileList) => {
+      const file = files[0];
+      if (!file) {
+        return;
+      }
+
+      const MAX_XML_SIZE = 1 * 1024 * 1024;
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx]: We should move it to const file

File: openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx [hunk 3]
```diff
@@ -249,6 +310,78 @@ const SSOConfigurationFormRJSF = ({
...
+      if (file.size > MAX_XML_SIZE) {
+        showErrorToast(t('message.file-size-exceeded', { size: '1 MB' }));
+
+        return;
+      }
+
+      const updateIdpFields = (fields: {
+        entityId: string | undefined;
+        ssoLoginUrl: string | undefined;
+        idpX509Certificate: string | undefined;
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx]: we can make it optional, else we have to send undefined

File: openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx (hunk 2 above)
Comment (Rohit0301) [reply openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx]: Done

File: openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx (hunk 3 above)
Comment (Rohit0301) [reply openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx]: Done

File: openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx [hunk 4]
```diff
@@ -366,9 +366,9 @@ const SSOConfigurationFormRJSF = ({
             setMetadataUploadStatus('success');
           } catch {
             updateIdpFields({
-              entityId: undefined,
-              ssoLoginUrl: undefined,
-              idpX509Certificate: undefined,
+              entityId: '',
+              ssoLoginUrl: '',
+              idpX509Certificate: '',
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx:371]: Could you please clarify the reason for making this change?

Also, wherever this value is being used, do we have proper checks in place to handle empty cases and prevent any unexpected behavior?

Comment (Rohit0301) [reply openmetadata-ui/src/main/resources/ui/src/components/SettingsSso/SSOConfigurationForm/SSOConfigurationForm.tsx:371]: When user upload a valid xml then we populates the data in the correct fields, but if user uploads an invalid XML then we need to clear those fields. Also we have validations for those fields, so when we make them empty, validations checks will run
